fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailActivity.java

```diff
@@ -35,19 +32,20 @@
 import com.github.dedis.popstellar.ui.qrcode.ScanningAction;
 import com.github.dedis.popstellar.ui.wallet.LaoWalletFragment;
 import com.github.dedis.popstellar.utility.ActivityUtils;
-import com.google.android.gms.vision.barcode.BarcodeDetector;
 
 import java.util.Objects;
 import java.util.function.Supplier;
 
+import dagger.hilt.android.AndroidEntryPoint;
+
+@AndroidEntryPoint
 public class LaoDetailActivity extends AppCompatActivity {
 
   private static final String TAG = LaoDetailActivity.class.getSimpleName();
   private LaoDetailViewModel mViewModel;
 
   public static LaoDetailViewModel obtainViewModel(FragmentActivity activity) {
-    ViewModelFactory factory = Injection.provideViewModelFactory(activity.getApplication());
-    return new ViewModelProvider(activity, factory).get(LaoDetailViewModel.class);
+    return new ViewModelProvider(activity).get(LaoDetailViewModel.class);
```

Comment:
Is it possible (and wouldn't it be better) to directly inject viewmodel and remove the function obtainViewModel ?

Reply:
No, this is an android limitation. The ViewModel must be created using a provider.
And Hilt knows that. That's why the ViewModels are injected when provided by a provider.